wled00/wled.cpp

```diff
@@ -285,6 +285,31 @@ DEBUG_PRINTLN(F("Watchdog: disabled"));
 #endif
 }
 
+#if defined(WLED_ENABLE_BACKGROUND)
```

Comment:
Can you move these two functions (background main loop, and creation of the new task) into the usermod?
I think its not strictly needed that this is added into the WLED core. You can achieve the same thing in the "setup" function of your main UsermodBackground:Setup. Similar to what usermods/audioreactive does.

https://github.com/Aircoookie/WLED/blob/45d7e664884a5d561435fa9a915f8ea96323a650/usermods/audioreactive/audio_reactive.h#L1173

https://github.com/Aircoookie/WLED/blob/45d7e664884a5d561435fa9a915f8ea96323a650/usermods/audioreactive/audio_reactive.h#L1356
https://github.com/Aircoookie/WLED/blob/45d7e664884a5d561435fa9a915f8ea96323a650/usermods/audioreactive/audio_reactive.h#L1379-L1405

Reply:
I will look at audio reactive for sure. I am not trying to reinvent the wheel, but make writing usermods easier. By adding backgroundLoop to wled.cpp and the other core files, all devs need to do is implement the backgroundLoop function. Also this strategy can support multiple usermods all with main and background loops. That said, it is never my goal to reinvent the wheel, just make the usermod experience easier for new devs. 

The stack size is large, but settable via the  BACKGROUND_STACK_SIZE setting.